Source/Scene/QuadtreePrimitive.js

```diff
@@ -489,210 +562,393 @@ define([
-        // some loading.
-        //
-        // So our load priority is (from high to low):
-        // 1. Tiles that we _would_ render, except that they're not sufficiently loaded yet.
-        //    Ideally this would only include tiles that we've already determined to be visible,
-        //    but since we don't have reliable visibility information until a tile is loaded,
-        //    and because we (currently) must have all children in a quad renderable before we
-        //    can refine, this pretty much means tiles we'd like to refine to, regardless of
-        //    visibility. (high)
-        // 2. Tiles that we're rendering. (medium)
-        // 3. All other tiles. (low)
-        //
-        // Within each priority group, tiles should be loaded in approximate near-to-far order,
-        // but currently they're just loaded in our traversal order which makes no guarantees
-        // about depth ordering.
+        var camera = frameState.camera;
+        primitive._neededPositions[0] = camera.positionCartographic;
+        var cameraFrameOrigin = Matrix4.getTranslation(camera.transform, cameraOriginScratch);
+        primitive._neededPositions[1] = primitive.tileProvider.tilingScheme.ellipsoid.cartesianToCartographic(cameraFrameOrigin, primitive._neededPositions[1]);
 
         // Traverse in depth-first, near-to-far order.
         for (i = 0, len = levelZeroTiles.length; i < len; ++i) {
             tile = levelZeroTiles[i];
             primitive._tileReplacementQueue.markTileRendered(tile);
             if (!tile.renderable) {
-                if (tile.needsLoading) {
-                    primitive._tileLoadQueueHigh.push(tile);
-                }
+                queueTileLoad(primitive, primitive._tileLoadQueueHigh, tile, frameState);
                 ++debug.tilesWaitingForChildren;
-            } else if (tileProvider.computeTileVisibility(tile, frameState, occluders) !== Visibility.NONE) {
-                visitTile(primitive, frameState, tile);
             } else {
-                if (tile.needsLoading) {
-                    primitive._tileLoadQueueLow.push(tile);
-                }
-                ++debug.tilesCulled;
+                visitIfVisible(primitive, tile, tileProvider, frameState, occluders, false, rootTraversalDetails[i]);
             }
         }
+
+        primitive._lastSelectionFrameNumber = frameNumber;
+    }
+
+    function queueTileLoad(primitive, queue, tile, frameState) {
+        if (!tile.needsLoading) {
+            return;
+        }
+
+        if (primitive.tileProvider.computeTileLoadPriority !== undefined) {
+            tile._loadPriority = primitive.tileProvider.computeTileLoadPriority(tile, frameState);
+        }
+        queue.push(tile);
     }
 
-    function visitTile(primitive, frameState, tile) {
+    function TraversalDetails() {
+        this.allAreRenderable = true;
+        this.anyWereRenderedLastFrame = false;
+        this.notYetRenderableCount = 0;
```

Comment:
I can see that these seem like a significant part of the traversal strategy. Do you think it deserves some doc comment here? It's unclear for example what it keeps counting `notYetRenderableCount` for.